src/high-level/matrix.lisp

```diff
@@ -379,6 +384,17 @@ If fast is t then just change layout. Fast can cause problems when you want to m
         (loop :for i :below rows
               :collect (tref matrix i i))))))
 
+(defgeneric block-diag (matrix)
```

Comment:
Same thing here -- seems like you have in mind specifically 2x2 block decompositions.